src/api.rs

```diff
@@ -124,19 +149,23 @@ impl GitHubApi {
             .context("Error: Failed to fetch matching pull request")
     }
 
-    pub async fn get_user_role(&self, user: &str) -> Result<String> {
-        // TODO change to using builtins ones implemented
+    pub async fn get_user_permission(&self, user: &str) -> Result<RepoPermission> {
+        // TODO change to using builtins ones implemented (affiliation parameter)
         let res = self
             .octocrab
-            .get::<UserRole, &str, ()>(
+            .get::<Vec<RawRepoPermission>, &str, ()>(
                 &format!(
-                    "/repos/{}/{}/collaborators/{user}/permission",
+                    "/repos/{}/{}/collaborators?affiliation=direct",
                     self.owner, self.repo
                 ),
                 None::<&()>,
             )
             .await?;
-        Ok(res.role_name)
+        res.into_iter()
+            .filter(|r| r.login.eq(user))
+            .flat_map(|r| RepoPermission::from_str(&r.role_name))
+            .next()
+            .ok_or(anyhow!("User {user} has no permissions specified"))
```

Comment:
Is it intentional that errors in `from_str` are ignored instead of returning them?

```suggestion
        res.into_iter()
            .find(|r| r.login.eq(user))
            .ok_or(anyhow!("User {user} has no permissions specified"))
            .and_then(|r| RepoPermission::from_str(&r.role_name))
```

Reply:
done